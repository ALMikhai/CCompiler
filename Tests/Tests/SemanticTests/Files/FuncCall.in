int foo(int a) {
    return a;
}

int f() {
    return foo(1);
}