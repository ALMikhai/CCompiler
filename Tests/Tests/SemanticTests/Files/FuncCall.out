\-TRANSLATIONUNIT
  |-FUNC
  | |-DECLSPEC
  | | |-INT
  | |-INITDECLARATOR
  | | \-DECLARATOR
  | |   \-FUNC
  | |     |-foo
  | |     \-PARAMLIST
  | |       \-PARAMDECLARATOR
  | |         |-DECLSPEC
  | |         | |-INT
  | |         \-DECLARATOR
  | |           \-a
  | \-COMPOUNDSTAT
  |   \-STATLIST
  |     \-RETURN
  |       \-a
  \-FUNC
    |-DECLSPEC
    | |-INT
    |-INITDECLARATOR
    | \-DECLARATOR
    |   \-FUNC
    |     |-f
    |     \-EMPTY
    \-COMPOUNDSTAT
      \-STATLIST
        \-RETURN
          \-()
            |-foo
            \-EXPLIST
              \-1

Structs { }
Symbols {
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: printf
	FUNC returning VOID
	Arguments{
		STRING :: s
	} :: scanf
	FUNC returning INT
	Arguments{
		INT :: a
	}
	Structs { }
	Symbols { } :: foo
	FUNC returning INT
	Arguments{ }
	Structs { }
	Symbols { } :: f
}
